src/devices/cpu/mcs51/mcs51.h

```diff
@@ -182,7 +182,7 @@ class mcs51_cpu_device : public cpu_device
 	void clear_current_irq();
 	uint8_t r_acc();
 	uint8_t r_psw();
-	offs_t external_ram_iaddr(offs_t offset, offs_t mem_mask);
+	virtual offs_t external_ram_iaddr(offs_t offset, offs_t mem_mask);
```

Comment:
Was this intentional?  Just seems a bit out-of-place given the nature of the rest of the change.

Reply:
it was originally, as the axc51 doesn't use P2 for the high bits best we can tell.

if the ds5002fp is going to be cleaned up it's needed anyway however, as half the logic in the existing function is behind a check for if the CPU is a ds5002fp, when really the ds5002 should just have an override for it.